iib/workers/tasks/utils.py

```diff
@@ -19,11 +19,141 @@
+    :param str distribution_scope: the scope for distribution
+        of the index image, defaults to ``None``.
+    :param str source_from_index: the pull specification of the container image
+        containing the index that will be used
+        as a base of the merged index image.
+    :param str target_index: the pull specification of the container image
+        containing the index whose new data will be added
+        to the merged index image.
+    :param dict binary_image_config: the dict of config required to
+        identify the appropriate ``binary_image`` to use.
+    """
+
+    _attrs = [
+        "_binary_image",
+        "distribution_scope",
+        "source_from_index",
+        "target_index",
+        "binary_image_config",
+    ]
+    __slots__ = _attrs
+
+    def __init__(self, **kwargs):
+        """
+        Request config __init__.
+
+        Do not use this directly, use subclasses instead.
+        :Keyword Arguments:
+            See `_attrs` to check accepted keyword arguments.
+        """
+        for key in self.__slots__:
+            setattr(self, key, None)
+        for key, val in kwargs.items():
+            setattr(self, key, kwargs[key])
+
+    def __eq__(self, other):
+        if type(self) == type(other) and [getattr(self, x) for x in self.__slots__] == [
+            getattr(self, x) for x in self.__slots__
+        ]:
+            return True
+        return False
+
+    def binary_image(self, index_info, distribution_scope):
+        """Get binary image based on self configuration, index image info and distribution scope."""
+        if not self._binary_image:
+            binary_image_ocp_version = index_info['ocp_version']
+            return get_binary_image_from_config(
+                binary_image_ocp_version, distribution_scope, self.binary_image_config
+            )
+        return self._binary_image
+
+
+class RequestConfigAddRm(RequestConfig):
+    """Request config for add and remove operations.
+
+    :param str overwrite_from_index_token: the token used for overwriting the input
+        ``from_index`` image. This is required for
+        non-privileged users to use ``overwrite_from_index``.
+        The format of the token must be
+        in the format "user:password".
+    :param str from_index: the pull specification of the container image
+        containing the index that the index image build
+        will be based from.
+    :param list add_arches: the list of arches to build in addition to the
+        arches ``from_index`` is currently built for;
+        if ``from_index`` is ``None``, then this is used as the list of arches
+        to build the index image for
+    :param list bundles: the list of bundles to create the
+        bundle mapping on the request
+    """
+
+    _attrs = RequestConfig._attrs + [
+        "overwrite_from_index_token",
+        "from_index",
+        "add_arches",
+        "bundles",
```

Comment:
Just noticed `operators` for `rm` is missing here

Reply:
"operators" are not needed for prepare_request_for_build

Reply:
Right, but it makes the code confusing if you create a config class for the Add and Rm request and it does not contain all the attributes a typical Add/Rm request would consist of. I think the classes should have everything and then we use whatever we want fo prepare_request_for_build